### ISSUE TITLE
HAProxy-405 Not Allowed-POST Method

### ISSUE THREAD

**Issue by git4suvendu** [haproxytech/kubernetes-ingress#119]
Hi,

I have an application (User Reg) written on Spring Boot which is running GCP Kubernetes cluster. The application inserts data into MySQL running on a different pod on the same cluster.
Using PostMan to call the API with the JSON body; its POST method.

HAProxy Ingress controller is running in the cluster and path based fanout ingress has been created which is pointing to a HTTP (L7) load balancer. GCP Cloud DNS is resolving the host name and point to loadbalancer.

I am expecting when I will call the API from PostNam the JSON Body data will flow like below and will be inserted into the database. 
PostMan -->Cloud DNS --> HTTP Loadbalancer --> HAProxy Ingress Controller --> User Reg Application --> MySQL

But I am getting 405 Not Allowed error. **Do I need to change any settings in the HAProxy so that it allows the POST method? Or anything else?**
 
Just a note, if I dont use Ingress Controller and expose the User Reg service directly to L4 loadbalancer then it works fine. Also, without the hostname in the fanout ingress yaml file if I deploy then it creates a L7 loadbalancer and the User Reg API can be called from PostMan with JSON body. 

Ingress YAML File
---------------
![fanout-ingress-yaml](https://user-images.githubusercontent.com/46852399/69505322-3c67e280-0f21-11ea-8a3f-77c47b9b3d49.JPG)

Describe Ingress
-------------------
![ingress-fanout-describe](https://user-images.githubusercontent.com/46852399/69505323-3d007900-0f21-11ea-9382-080354476237.JPG)

All Running PODS in Cluster
--------------------------------
![PODS](https://user-images.githubusercontent.com/46852399/69505324-3d007900-0f21-11ea-92af-d9bfb7f2f881.JPG)

All Running Services in Cluster
-----------------------------------
![services](https://user-images.githubusercontent.com/46852399/69505325-3d007900-0f21-11ea-8a64-daa27d37fc7c.JPG)

UserReg application Controller
--------------------------------
![User-Reg-Controller](https://user-images.githubusercontent.com/46852399/69505382-8fda3080-0f21-11ea-8f00-f1cff4c59e6b.JPG)

PostMan API call and request body and response body
----------------------------------------------------------
![image](https://user-images.githubusercontent.com/46852399/69505493-1a229480-0f22-11ea-83a4-9b02c021a1c3.png)

**Comment by bedis** [comment]
Could it be related to CORS? Anything to setup on your app related to this topic?

**Comment by git4suvendu** [comment]
> Could it be related to CORS? Anything to setup on your app related to this topic?

No, I haven't anything related to CORS in the app. Should I?
I am trying to hit the URL (demo.apps.mydomain...) from my desktop over the internet. So do I need to register the domain mydomain.com before I hit over the internet?

**Comment by bedis** [comment]
Well this seem to be your app refusing the POST request, so I guess the fix should happen in your app.
Just to double check, can you enable logging in the controller and share here the relevant log line?

**Comment by git4suvendu** [comment]
> Well this seem to be your app refusing the POST request, so I guess the fix should happen in your app.
> Just to double check, can you enable logging in the controller and share here the relevant log line?

I can do. What I did now is removed the hostname(demo.apps.mydomain...) from the ingress yaml file and accessing the app using loadbalancer ip from PostMan over the internet. Guess what..it works fine !!! Then how come its a CORS issue? Why the hostname is having an issue then?

Updated Ingress YAML
---------------------------
![image](https://user-images.githubusercontent.com/46852399/69531643-1fa4cc80-0f6c-11ea-93a4-ce46afdac35d.png)
